geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSet.java

```diff
@@ -94,6 +102,34 @@ public RedisSet(int size) {
     return diff;
   }
 
+  private static int setOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      Set<byte[]> diff) {
+    if (regionProvider.getRedisData(destinationKey).getType() != REDIS_SET) {
+      regionProvider.getDataRegion().remove(destinationKey);
+    }
+
+    boolean destinationKeyExists = regionProvider.getRedisData(destinationKey).exists();
+    if (diff.isEmpty()) {
+      if (destinationKeyExists) {
+        regionProvider.getDataRegion().remove(destinationKey);
+      }
+      return 0;
+    }
+
+    if (destinationKeyExists) {
+      RedisSet destinationSet = regionProvider.getTypedRedisData(REDIS_SET, destinationKey, false);
+
+      destinationSet.members.clear();
```

Comment:
If you know "diff" is a MemberSet then instead of clear and addAll you can just do "destinationSet.members = diff".

Reply:
something missing here is a call to persistNoDelta